src/ActionBuilder/ProductType/ChangeAttributeLabel.php

```diff
@@ -41,15 +41,25 @@ public function createUpdateActions(
         array $oldData,
         $sourceObject
     ): array {
-        list(, $attrIndex) = $this->getLastFoundMatch();
+        $oldLabels = [];
+        foreach ($oldData['attributes'] as $attribute) {
+            $oldLabels[$attribute['name']] = $attribute['label'];
+        }
 
         $actions = [];
+        foreach ($sourceObject->getAttributes()->toArray() as $attribute) {
+            $attributeName = $attribute['name'];
+
+            if (!array_key_exists($attributeName, $oldLabels)) {
+                continue;
+            }
 
-        if ((@$oldData['attributes'][$attrIndex]) && ($def = $sourceObject->getAttributes()->getAt($attrIndex))) {
```

Comment:
This does not work when removing attributes because the index are different for oldData and source.